ngi_pipeline/engines/utils.py

```diff
@@ -0,0 +1,62 @@
+        return False
+    elif charon_sr.get('alignment_status') == 'FAILED' and not analysis_object.restart_failed_jobs:
+        LOG.info("seqrun {}/{}/{}/{} analysis has failed, but no restart_failed flag was given, skipping.".format(analysis_object.project.project_id, sample.name, libprep.name, seqrun.name))
+        return False
+    else:
+        return True
+
+def handle_libprep_status(analysis_object, libprep, charon_reported_status):
+    if charon_reported_status == 'FAILED':
+        LOG.info("libprep {}/{}/{} is marked as failed, skipping all of its seqruns.".format(analysis_object.project.project_id, sample.name, libprep.name))
+        return False
+    else:
+        return True
+
+def handle_sample_status(analysis_object, sample, charon_reported_status):
+    """ returns true of false wether the sample should be analyzed"""
+    if charon_reported_status == "UNDER_ANALYSIS":
+        if not analysis_object.restart_running_jobs:
+            error_text = ('Charon reports seqrun analysis for project "{}" '
+                          '/ sample "{}" does not need processing (already '
+                          '"{}")'.format(analysis_object.project, sample, charon_reported_status))
+            LOG.error(error_text)
+            if not analysis_object.config.get('quiet'):
+                mail_analysis(project_name=analysis_object.project.name, sample_name=sample.name,
+                              engine_name=analysis_module.__name__,
+                              level="ERROR", info_text=error_text)
+            return False 
+    elif charon_reported_status == "ANALYZED":
+        if not analysis_object.restart_finished_jobs:
+            error_text = ('Charon reports seqrun analysis for project "{}" '
+                          '/ sample "{}" does not need processing (already '
+                          '"{}")'.format(analysis_object.project, sample, charon_reported_status))
+            LOG.error(error_text)
+            if not analysis_object.config.get('quiet') and not analysis_object.config.get('manual'):
+                mail_analysis(project_name=analysis_object.project.name, sample_name=sample.name,
+                              engine_name=analysis_module.__name__,
+                              level="ERROR", info_text=error_text)
+            return False 
+    elif charon_reported_status == "FAILED":
+        if not analysis_object.restart_failed_jobs:
+            error_text = ('FAILED:  Project "{}" / sample "{}" Charon reports '
+                          'FAILURE, manual investigation needed!'.format(analysis_object.project, sample))
+            LOG.error(error_text)
+            if not analysis_object.config.get('quiet'):
+                mail_analysis(project_name=analysis_object.project.name, sample_name=sample.name,
+                              engine_name=analysis_module.__name__,
+                              level="ERROR", info_text=error_text)
+            return False 
+    else:
```

Comment:
let me see if I understand well.
If `charon_reported_status ` is `ANALYZED` but I want to rerun the sample anyway (i.e., `analysis_object.restart_finished_jobs = TRUE`, then line 41 evaluates to `FALSE`. So I will not enter in any other else/if section and this function will return `None`, therefore in piper_ngi/launchers.py inline 66 I will evaluate to `TRUE` and do not restart the analysis....